File: scraper/check_images.py
import os
from dotenv import load_dotenv
from supabase import create_client, Client

load_dotenv()

SUPABASE_URL = "https://dyjcskzwgztfzrafefxa.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImR5amNza3p3Z3p0ZnpyYWZlZnhhIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjM5MTI2ODIsImV4cCI6MjA3OTQ4ODY4Mn0.h24P81bLHQaRDOvSwsRT6c9GiYzfpYcSe6D4xmzLpdc"

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

response = supabase.table('noticias').select('titulo, imagen_principal_url, medio_comunicacion').order('fecha_publicacion', desc=True).limit(10).execute()

print(f"{'MEDIO':<20} | {'TITULO':<40} | {'IMAGEN URL'}")
print("-" * 100)
for noticia in response.data:
    titulo = (noticia['titulo'][:37] + '...') if len(noticia['titulo']) > 37 else noticia['titulo']
    print(f"{noticia['medio_comunicacion']:<20} | {titulo:<40} | {noticia['imagen_principal_url']}")

File: frontend/check-images.js

const https = require('https');

const supabaseUrl = 'https://unitbhhizdvtvsmplhon.supabase.co';
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InVuaXRiaGhpemR2dHZzbXBsaG9uIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjM2NDc1ODcsImV4cCI6MjA3OTIyMzU4N30.uQTs1Ax6CtDj_LO0osCPm1JjEPLj2BHjk5H9LLz8qms';

const url = `${supabaseUrl}/rest/v1/noticias?select=id,titulo,imagen_principal,url_original&order=fecha_publicacion.desc&limit=5`;
const options = {
    method: 'GET',
    headers: {
        'apikey': supabaseKey,
        'Authorization': `Bearer ${supabaseKey}`
    }
};

console.log('Fetching news images...');

const req = https.request(url, options, (res) => {
    let data = '';
    res.on('data', (chunk) => {
        data += chunk;
    });

    res.on('end', () => {
        try {
            const news = JSON.parse(data);
            console.log(`Fetched ${news.length} news items.`);
            news.forEach(n => {
                console.log(`\nID: ${n.id}`);
                console.log(`Title: ${n.titulo}`);
                console.log(`Image URL: ${n.imagen_principal}`);
                console.log(`Original URL: ${n.url_original}`);
            });
        } catch (e) {
            console.error('Error parsing response:', e);
            console.log('Raw body:', data);
        }
    });
});

req.on('error', (e) => {
    console.error('Request error:', e);
});

req.end();
